### DISCUSSION TITLE
Enable automatic panning of map when the node being dragged reaches edge of the screen

### DISCUSSION THREAD

**Discussion by Anand2489** [visgl/deck.gl#6370] [Q&A] [answered]
I'm rendering nodes(vertices) using IconLayer with dragging enabled on these nodes. When user isn't dragging any node, state of my controller is 
```jsx
<DeckGL
     controller={true}
     ...
/>  
```

While dragging a node, state of my controller is 
```jsx
<DeckGL
     controller={false}
     ...
/>  
```
On drag end, state of my controller is 
```jsx
<DeckGL
     controller={true}
     ...
/>  
```

**How can I enable automatic panning of map when the node being dragged reaches edge of the screen ?**

@Pessimistress @ibgreen ^^

**Comment by Pessimistress** [comment, marked as answer]
Demo: https://codepen.io/Pessimistress/pen/xxLBYPz

- On drag event, check `event.x` and `event.y` to see if the pointer is within a threshold from the border
- Use the method `viewport.panByPosition(coordinate, pixel)` to get the view state needed to pan `coordinate` (world) to `pixel` (screen)
- Update the camera by setting `initialViewState`
